src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,107 @@
+import { cpUtils } from "../../../utils/cpUtils";
+import { nonNullProp } from '../../../utils/nonNull';
+import { openUrl } from '../../../utils/openUrl';
+import { IJavaProjectWizardContext } from '../../createNewProject/javaSteps/IJavaProjectWizardContext';
+import { IDotnetFunctionWizardContext } from '../dotnetSteps/IDotnetFunctionWizardContext';
+import { IFunctionWizardContext } from "../IFunctionWizardContext";
+
+export class OpenAPICreateStep extends AzureWizardExecuteStep<IFunctionWizardContext> {
+    public priority: number = 220;
+
+    public static async createStep(context: IActionContext): Promise<OpenAPICreateStep> {
+        await validateAutorestInstalled(context);
+        return new OpenAPICreateStep();
+    }
+
+    public async execute(wizardContext: IFunctionWizardContext & IJavaProjectWizardContext & IDotnetFunctionWizardContext): Promise<void> {
+        const uris: Uri[] = await this.askDocument();
+        const uri: Uri = uris[0];
+        const args: string[] = [];
+
+        args.push(`--input-file:${cpUtils.wrapArgInQuotes(uri.fsPath)} --version:3.0.6320`);
+
+        switch (wizardContext.language) {
+            case ProjectLanguage.TypeScript:
+                args.push('--azure-functions-typescript');
+                args.push('--no-namespace-folders:True');
+                break;
+            case ProjectLanguage.CSharp:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'namespace')}`);
+                args.push('--azure-functions-csharp');
+                break;
+            case ProjectLanguage.Java:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'javaPackageName')}`);
+                args.push('--azure-functions-java');
+                break;
+            case ProjectLanguage.Python:
+                args.push('--azure-functions-python');
+                args.push('--no-namespace-folders:True');
+                args.push('--no-async');
+                break;
+            default:
+                throw new Error(localize('notSupported', 'Not a supported language. We currently support C#, Java, Python, and Typescript'));
+                break;
+        }
+
+        args.push('--generate-metadata:false');
+        args.push(`--output-folder:${wizardContext.projectPath}`);
+
+        ext.outputChannel.appendLog(localize('statutoryWarning', 'Using the plugin could overwrite your custom changes to the functions.'));
+        const title: string = localize('generatingFunctions', 'Generating from OpenAPI...Check [output window](command:{0}) for status.', ext.prefix + '.showOutputChannel');
+
+        await window.withProgress({ location: ProgressLocation.Notification, title }, async () => {
+            await cpUtils.tryExecuteCommand(ext.outputChannel, undefined, 'autorest', ...args);
+        });
+    }
+
+    public shouldExecute(): boolean {
+        return true;
+    }
+
+    public async askDocument(): Promise<Uri[]> {
```

Comment:
You need to create an `AzureWizardPromptStep` for this. There shouldn't be any prompts as a part of an "execute" step